aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -496,4 +553,22 @@ module aptos_framework::account {
         coin::destroy_mint_cap(mint_cap);
         let _bob = bob;
     }
+
+    #[test(alice = @0xa11ce)]
+    #[expected_failure(abort_code = 65548)]
+    public entry fun test_invalid_public_key(alice: signer) acquires Account, OriginatingAddress {
+        create_account(signer::address_of(&alice));
+        let test_public_key = vector::empty<u8>();
+        let test_signature = vector::empty<u8>();
+        rotate_authentication_key_ed25519(&alice, test_public_key, test_signature);
```

Comment:
Again, folks, **type safety**, please can we have it? By having vector<u8>'s everywhere you can confuse parameters as you've done here: the public key and the signature should be swapped.